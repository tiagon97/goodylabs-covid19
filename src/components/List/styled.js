import styled from 'styled-components';

export const StyledWrapper = styled.ul`
  list-style: none;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
`;

export const StyledItem = styled.li`
  font-size: 1.4rem;
  padding: 20px;
  margin: 2px;
  border: solid #ddd 2px;
  transition: 0.3s;
  border-radius: 5px;
  flex-basis: calc(20% - 4px);
  text-align: center;

  &:hover {
    background-color: #ddd;
    color: white;
    cursor: pointer;
    font-weight: 600;
    letter-spacing: 3px;
  }
`;
