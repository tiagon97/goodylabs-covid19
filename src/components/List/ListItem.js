import React from 'react';
import styled from 'styled-components';

const StyledItem = styled.li`
  padding: 20px;
  margin: 1px;
  border: solid #ddd 2px;
  flex-basis: calc(25% - 2px);
  transition: 0.3s;
  border-radius: 5px;

  :hover {
    background-color: #ddd;
    color: white;
  }
`;

const ListItem = ({ children }) => <StyledItem>{children}</StyledItem>;
export default ListItem;
