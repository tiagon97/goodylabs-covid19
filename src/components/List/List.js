import React from 'react';
import styled from 'styled-components';
import ListItem from './ListItem';

const StyledWrapper = styled.ul`
  list-style: none;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
`;

const List = ({ filteredCountries, clearFn }) => (
  <StyledWrapper>
    {filteredCountries.map((item) => {
      return (
        <ListItem clearFn={clearFn} key={item.CountryCode}>
          {item.Country}
        </ListItem>
      );
    })}
  </StyledWrapper>
);

export default List;
