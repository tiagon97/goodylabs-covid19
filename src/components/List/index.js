import React from 'react';

import ListItem from './ListItem';
import { StyledWrapper } from './styled';

const List = ({ filteredCountries, selectCountry }) => (
  <StyledWrapper>
    {filteredCountries.map((item) => {
      return (
        <ListItem selectCountry={selectCountry} key={item.CountryCode}>
          {item.Country}
        </ListItem>
      );
    })}
  </StyledWrapper>
);

export default List;
