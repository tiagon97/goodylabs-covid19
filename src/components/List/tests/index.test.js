/**
 *
 * Tests for List
 *
 */

import React from 'react';
import { render } from '@testing-library/react';

import List from '../index';

const defaultProps = {
  selectCountry: jest.fn(),
  filteredCountries: [
    {
      Country: 'Poland',
      CountryCode: 'PL',
      Date: '2020-05-24T11:44:37Z',
      NewConfirmed: 312,
      NewDeaths: 11,
      NewRecovered: 246,
      Slug: 'poland',
      TotalConfirmed: 20931,
      TotalDeaths: 993,
      TotalRecovered: 8977,
    },
  ],
};

describe('<List />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<List {...defaultProps} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<List {...defaultProps} />);
    expect(firstChild).toMatchSnapshot();
  });
});
