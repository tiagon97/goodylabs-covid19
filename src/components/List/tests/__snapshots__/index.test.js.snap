// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> Should render and match the snapshot 1`] = `
<ul
  class="sc-AxjAm jwWEQX"
>
  <li
    class="sc-AxirZ ggiwpy"
  >
    Poland
  </li>
</ul>
`;
